src/subcommands/add.go

```diff
@@ -1,184 +1,228 @@
 package subcommands
 
 import (
-	"encoding/gob"
 	"errors"
 	"fmt"
 	"io"
 	"os"
 	"path"
 	"path/filepath"
-	"time"
+
+	pb "accretional.com/semantifly/proto/accretional.com/semantifly/proto"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 const addedCopiesSubDir = "add_cache"
-const addedFile = "added.list"
-
-type AddedListEntry struct {
-	Name           string
-	URI            string
-	DataType       string
-	SourceType     string
-	TimeFirstAdded string
-}
-
-type AddCacheEntry struct {
-	AddedListEntry
-	TimeLastRefreshed string
-	Contents          []byte
-}
+const indexFile = "index.list"
 
 type AddArgs struct {
 	IndexPath  string
-	DataType   string
-	SourceType string
+	DataType   pb.DataType
```

Comment:
I think it's easier to test Add if we allow arbitrary string args and only convert to pb.DataType internally, but up to you whether to change this back or not

Reply:
I still feel we can keep it in the subcommand